### PR TITLE
rubocop implementation

### PR DESCRIPTION
**Pull request by scara82**
To be replaced by add_rubocop (https://github.com/hooroo/domus/pull/6)
Rubocop Implementation
Files fixes for Rubocop

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,56 +0,0 @@
...
-  Enabled: true
-Style/OneLineConditional:
-  Enabled: false
-Style/StderrPuts:
-  Exclude:
-    - "bin/*"
-Style/StringLiterals:
-  EnforcedStyle: double_quotes
-Style/TrailingCommaInLiteral:
-  EnforcedStyleForMultiline: consistent_comma
```
Comment (elle) [inline .rubocop.yml:56]: Is this file supposed to be removed?

File: config/environments/test.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 Rails.application.configure do
+
```
Comment (elle) [inline config/environments/test.rb:2]: This line is not needed.
You might want to add the following rule:
```
Layout/EmptyLinesAroundBlockBody:
  Description: checks if empty lines around the bodies of blocks
  Enabled: no_empty_lines
```

Which looking at the documentation should be the default: https://rubocop.readthedocs.io/en/latest/cops_layout/#layoutemptylinesaroundblockbody

File: spec/factories.rb [hunk 3]
```diff
@@ -1,2 +1,2 @@
-FactoryBot.define do
+FactoryBot.define do
```
Comment (elle) [inline spec/factories.rb:1]: There is an extra space after the `do`. Not sure what the rubocop rule is to check that

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -2,6 +2,9 @@
 ENV["RAILS_ENV"] ||= "test"
 require File.expand_path("../../config/environment", __FILE__)
 
+abort("The Rails environment is running in production mode!") if Rails.env.production?
```
Comment (elle) [inline spec/rails_helper.rb:5]: This should be:
```
if Rails.env.production?
  abort("The Rails environment is running in production mode!") 
end
```

File: config/environments/production.rb [hunk 5]
```diff
@@ -52,7 +52,7 @@
   config.log_level = :debug
 
   # Prepend all log lines with the following tags.
-  config.log_tags = [:request_id]
+  config.log_tags = [ :request_id ]
```
Comment (elle) [inline config/environments/production.rb:55]: The removed line is the correct format

File: config/environments/production.rb [hunk 6]
```diff
@@ -21,7 +21,7 @@
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
-  config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
+  config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (elle) [inline config/environments/production.rb:24]: Double quotes

File: config/environments/development.rb [hunk 7]
```diff
@@ -13,12 +13,12 @@
...
 
   # Enable/disable caching. By default caching is disabled.
-  if Rails.root.join("tmp/caching-dev.txt").exist?
+  if Rails.root.join('tmp/caching-dev.txt').exist?
     config.action_controller.perform_caching = true
 
     config.cache_store = :memory_store
     config.public_file_server.headers = {
-      "Cache-Control" => "public, max-age=#{2.days.seconds.to_i}",
+      'Cache-Control' => "public, max-age=#{2.days.seconds.to_i}"
```
Comment (elle) [inline config/environments/development.rb:21]: Double quotes

File: config/environments/development.rb [hunk 8]
```diff
@@ -13,12 +13,12 @@
   config.consider_all_requests_local = true
 
   # Enable/disable caching. By default caching is disabled.
-  if Rails.root.join("tmp/caching-dev.txt").exist?
+  if Rails.root.join('tmp/caching-dev.txt').exist?
```
Comment (elle) [inline config/environments/development.rb:16]: Double quotes

File: config/environment.rb [hunk 9]
```diff
@@ -1,5 +1,5 @@
 # Load the Rails application.
-require_relative "application"
+require_relative 'application'
```
Comment (elle) [inline config/environment.rb:2]: Double quotes

File: config/boot.rb [hunk 10]
```diff
@@ -1,3 +1,3 @@
-ENV["BUNDLE_GEMFILE"] ||= File.expand_path("../Gemfile", __dir__)
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
 
-require "bundler/setup" # Set up gems listed in the Gemfile.
+require 'bundler/setup' # Set up gems listed in the Gemfile.
```
Comment (elle) [inline config/boot.rb:3]: Double quotes

File: config/application.rb [hunk 11]
```diff
@@ -1,4 +1,4 @@
-require_relative "boot"
+require_relative 'boot'
```
Comment (elle) [inline config/application.rb:1]: Double quotes

File: config.ru [hunk 12]
```diff
@@ -1,5 +1,5 @@
 # This file is used by Rack-based servers to start the application.
 
-require_relative "config/environment"
+require_relative 'config/environment'
```
Comment (elle) [inline config.ru:3]: Double quotes

File: app/mailers/application_mailer.rb [hunk 13]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: "from@example.com"
-  layout "mailer"
+  default from: 'from@example.com'
+  layout 'mailer'
```
Comment (elle) [inline app/mailers/application_mailer.rb:3]: Double quotes

File: Rakefile [hunk 14]
```diff
@@ -1,3 +1,6 @@
-require_relative "config/application"
+# Add your own tasks in files placed in lib/tasks ending in .rake,
+# for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
+
+require_relative 'config/application'
```
Comment (elle) [inline Rakefile:4]: Double quotes

File: Gemfile [hunk 15]
```diff
@@ -16,23 +16,21 @@ group :development, :test do
   gem "hirb"
   gem "pry-byebug"
   gem "rspec-rails", "3.6.1"
-  gem "rubocop"
```
Comment (elle) [inline Gemfile:19]: Rubocop is removed?

File: Gemfile [hunk 16]
```diff
@@ -16,23 +16,21 @@ group :development, :test do
...
 end
 
 group :development do
+  gem "web-console", ">= 3.3.0"
   gem "listen", ">= 3.0.5", "< 3.2"
   gem "spring"
   gem "spring-watcher-listen", "~> 2.0.0"
-  gem "web-console", ">= 3.3.0"
```
Comment (elle) [inline Gemfile:26]: Gem should be sorted alphabetically

**Review by elle — COMMENTED** [review]
It looks like rubicon has been undone in this :/
Maybe something happened along the way??

**Comment by elle** [comment]
s/rubicon/rubocop/

**Comment by scara82** [comment]
@elle So this was an old PR, you already review it, I made the changes and then I merged without the final approval. Earler today I realised that was the wrong procedure so I undone it.

I will redo also this one. Starting from scratch in a proper way. I'm sorry for the mess, I will update you once all the proper PRs will be out!

But I'm learning the right procedure :)

Thanks!

**Pull request closed without merging**